contentcuration/contentcuration/tasks.py

```diff
@@ -56,6 +57,37 @@
 # runs the management command 'exportchannel' async through celery
 
 
+@task(bind=True, name="move_nodes_task")
+def move_nodes_task(
+    self,
+    user_id,
+    channel_id,
+    target_id,
+    node_id,
+    position="last-child",
+):
+    node = ContentNode.objects.get(id=node_id)
+    target = ContentNode.objects.get(id=target_id)
+
+    moved = False
+    attempts = 0
+
+    while not moved and attempts < 10:
+        try:
+            node.move_to(
+                target,
+                position,
+            )
+            moved = True
+        except OperationalError as e:
+            if "deadlock detected" in e.args[0]:
+                pass
+            else:
+                break
+
+    return {"changes": [generate_update_event(node.pk, CONTENTNODE, {"parent": node.parent_id})]}
```

Comment:
Should we handle the case where all 10 attempts fail, e.g. sending a Sentry error and returning some sort of sync error?

Reply:
We could log an additional sentry error, yeah, I'll add that in.

We don't currently have a good mechanism for reporting errors to async processed changes to the frontend, hence the simple revert here.

Reply:
Have added it in.

The simplest overall solution here might be to make the 'sync' endpoint processing completely async, and just have it store an event queue that can then get processed asynchronously, but that will require a few more changes.